mlir/lib/Conversion/ArmSMEToSCF/ArmSMEToSCF.cpp

```diff
@@ -506,6 +496,7 @@ struct TileVectorPrintOpConversion : public OpRewritePattern<vector::PrintOp> {
       rewriter.setInsertionPointToStart(forOp.getBody());
       // Extract the current row from the tile.
       Value rowIndex = forOp.getInductionVar();
+      // FIXME: Forward tile IDs.
```

Comment:
It's because `vector.print` is not a `ArmSMETileOp`, so unlike the other ops this would require looking at the use-def chain of the tile.

Reply:
As it's only for debug, I didn't think it was crucial :smile:

Reply:
ok thanks for the clarification. I thought this was currently used in the integration tests?

Reply:
It's used it some, but not all. The simple workaround is to do `-allocate-arm-sme-tiles` after `-convert-arm-sme-to-scf` (since this lowers it to `ArmSMETileOps`).

Reply:
Done :+1:

Reply:
Another solution is just to move this to `convert-vector-to-arm-sme`, which I think is fine (as we already generate loops for arith ops in there).

Reply:
(That can be a follow up patch though)

Reply:
This got me thinking. The tile allocation works by assigning tile IDs for the root operations that implement the `ArmSMETileOpInterface` and allocate tiles. An operation allocates a tile if it implements `getAllocatedTileType`, this includes:
```mlir
arm_sme.get_tile
arm_sme.zero
arm_sme.tile_load
arm_sme.outerproduct // (if no accumulator is specified)
```
the tile ID is then forwarded to any uses of the root that also implement the `ArmSMETileOpInterface`. Is this constraint necessary? If the tile ID could be forwarded to any use (i.e. non-ArmSME ops) could it fix this problem?

An alternative of course is to mirror ops such as `vector.print` in the ArmSME dialect with ops that implement this interface.

Reply:
I think the correct fix here is just to more the vector.print lowering into `-convert-vector-to-arm-sme` as it's a vector op lowering, not an ArmSME op lowering.

Reply:
Which is how other similar tile ops are lowered (e.g. vector.splat, vector.broadcast, etc). Using an interface allows setting/getting the tile IDs via methods (and validating them), it also avoids possible conflicts with other dialects (and prevents the IR from getting very messy.

Reply:
> I think the correct fix here is just to more the vector.print lowering into `-convert-vector-to-arm-sme` as it's a vector op lowering, not an ArmSME op lowering (no extra op required).

Ok, I think that's fine for now. As you mentioned, this doesn't have to be part of this patch. We may also have to revisit loop materialization in `-convert-vector-to-arm-sme`. When loads/stores were first implemented loop materialization was done during lowering to intrinsics, `-convert-arm-sme-to-scf` was introduced to break that up such that the intrinsic lowerings were 1-1. It's not quite the same when lowering Vector ops to ArmSME, but there may be value in getting the granularity right, that could necessitate more ops in ArmSME however, so some cost there at least.